includes/admin/class-wc-rest-payments-reader-controller.php

```diff
@@ -62,4 +100,69 @@ public function get_summary( $request ) {
 		return rest_ensure_response( $summary );
 	}
 
+	/**
+	 * Renders HTML for a print receipt
+	 *
+	 * @param WP_REST_Request $request Full data about the request.
+	 * @return WP_HTTP_RESPONSE|WP_Error
+	 */
+	public function generate_print_receipt( $request ) {
+		try {
+			$payment_intent = $this->api_client->get_intent( $request->get_param( 'payment_id' ) );
+
+			if ( 'succeeded' !== $payment_intent->get_status() ) {
+				return rest_ensure_response( new WP_Error( 'generate_print_receipt_error', __( 'Invalid payment', 'woocommerce-payments' ), [ 'status' => 500 ] ) );
+			}
+
+			$charge = $this->api_client->get_charge( $payment_intent->get_charge_id() );
+		} catch ( API_Exception $e ) {
+			return rest_ensure_response( new WP_Error( 'generate_print_receipt_error', $e->getMessage(), [ 'status' => $e->get_http_code() ] ) );
+		}
+
+		$order = wc_get_order( $charge['order']['number'] );
+
+		if ( ! $order ) {
+			return rest_ensure_response( new WP_Error( 'generate_print_receipt_error', __( 'Order not found', 'woocommerce-payments' ), [ 'status' => 500 ] ) );
+		}
+
+		try {
+			$settings = [
+				'business_name' => $this->wcpay_gateway->get_option( 'account_business_name' ),
+				'support_info'  => [
+					'address' => $this->wcpay_gateway->get_option( 'account_business_support_address' ),
+					'phone'   => $this->wcpay_gateway->get_option( 'account_business_support_phone' ),
+					'email'   => $this->wcpay_gateway->get_option( 'account_business_support_email' ),
+				],
+			];
+		} catch ( Exception $e ) {
+			return rest_ensure_response( new WP_Error( 'generate_print_receipt_error', __( 'There was a problem retrieving merchant settings.', 'woocommerce-payments' ), [ 'status' => 500 ] ) );
+		}
+
+		/**
+		 * WP_REST_Server will convert the response data to JSON prior to output it.
+		 * Using this filter to prevent it, and output the data from WP_HTTP_Responde instead.
```

Comment:
There's a small typo in `WP_HTTP_Responde`

Reply:
Thanks @KarlisJ!